### PR TITLE
crimson/seastore: remove unused method

### PR DESCRIPTION
**Pull request by Aran85**
Signed-off-by: Zengran Zhang <zhangzengran@sangfor.com.cn>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/os/seastore/cached_extent.h [hunk 1]
```diff
@@ -553,7 +553,8 @@ class ExtentIndex {
   template <typename T>
   void remove(T &l) {
     for (auto &ext : l) {
-      extent_index.erase(l);
+      static_assert(std::is_same_v<decltype(ext), CachedExtent::index::value_type>);
+      extent_index.erase(ext);
```
Comment (tchaikov) [inline src/crimson/os/seastore/cached_extent.h:557]: @athanatos i'd suggest just drop this method. what do you think?

**Comment by tchaikov** [comment]
@athanatos ping?

File: src/crimson/os/seastore/cached_extent.h (hunk 1 above)
Comment (athanatos) [reply src/crimson/os/seastore/cached_extent.h:557]: Yeah, if it's unused just remove it.

**Comment by tchaikov** [comment]
@Aran85 could you repurpose this change to drop this method since it is not used yet?

**Comment by tchaikov** [comment]
```
Building remotely on 172.21.1.35+slave-ubuntu01 (python3 small sepia x86_64 focal bionic huge xenial trusty amd64) in workspace /home/jenkins-build/build/workspace/ceph-api
The recommended git tool is: git
...
2021-05-24T01:53:14.644+0000 7ff64961e580 -1 error: monitor data filesystem reached concerning levels of available storage space (available: 4% 17 GiB)
```

**Comment by tchaikov** [comment]
fixed by #41492

**Review by tchaikov — APPROVED** [review]

**Pull request merged**